src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java

```diff
@@ -936,6 +985,166 @@ void testGetApprovedDatasetsWhenEmpty() {
 
   }
 
+  @Test
+  void testGetApprovedDatasetsForMultiDACElections() {
+    Date now = new Date();
+    // user with a mix of approved and unapproved datasets
+    User user = createUser();
+    User chairperson1 = createUser();
+    User chairperson2 = createUser();
+    libraryCardDAO.insertLibraryCard(user.getUserId(), user.getDisplayName(), user.getEmail(), user.getUserId(), now);
+
+    Dataset dataset1 = createDataset(true);
+    Dataset dataset2 = createDataset(true);
+    Dataset dataset3 = createDataset(true);
+    Dataset dataset4 = createDataset(true);
+
+    Timestamp timestamp = new Timestamp(now.getTime());
+
+    Dac dac1 = insertDac();
+    datasetDAO.updateDataset(dataset1.getDatasetId(), dataset1.getDatasetName(), timestamp, chairperson1.getUserId(), dac1.getDacId());
+    datasetDAO.updateDataset(dataset2.getDatasetId(), dataset2.getDatasetName(), timestamp, chairperson1.getUserId(), dac1.getDacId());
+    datasetDAO.updateDatasetApproval(true, Instant.now(), chairperson1.getUserId(), dataset1.getDatasetId());
+    datasetDAO.updateDatasetApproval(true, Instant.now(), chairperson1.getUserId(), dataset2.getDatasetId());
+
+
+    Dac dac2 = insertDac();
+    datasetDAO.updateDataset(dataset3.getDatasetId(), dataset3.getDatasetName(), timestamp, chairperson2.getUserId(), dac2.getDacId());
+    datasetDAO.updateDataset(dataset4.getDatasetId(), dataset4.getDatasetName(), timestamp, chairperson2.getUserId(), dac2.getDacId());
+    datasetDAO.updateDatasetApproval(true, Instant.now(), chairperson2.getUserId(), dataset3.getDatasetId());
+    datasetDAO.updateDatasetApproval(true, Instant.now(), chairperson2.getUserId(), dataset4.getDatasetId());
+
+    DarCollection darCollection = createDarCollectionWithDatasetsNewModel(user, List.of(dataset1, dataset2, dataset3, dataset4));
```

Comment:
Build a data access request (and related objects) with our new model of combining these into a single DAR.